recipes/newrelic/apm/php-agent/php.yml

```diff
@@ -29,20 +29,80 @@ inputVars:
 install:
   version: "3"
 
+  vars:
+    TMP_INSTALL_DIR:
+      sh: mktemp -d /tmp/newrelic-php-agent.XXXXXX
+
   env:
     NEW_RELIC_DISTRIBUTED_TRACING_ENABLED: true
 
   tasks:
     default:
       cmds:
-        - task: install_packages
         - task: create_shared
+        - task: detect_services
+        - task: install_packages
         - task: config_newrelic_ini
         - task: finalize_agent_config
         - task: restart_services
         - task: send_transaction
         - task: delete_shared
 
+    create_shared:
+      cmds:
+        - rm -rf {{.TMP_INSTALL_DIR}}
+        - mkdir -p {{.TMP_INSTALL_DIR}}
+        - echo "Created temporary directory {{.TMP_INSTALL_DIR}} for installation."
+
+    detect_services:
+      cmds:
+        - echo "Detect services."
+        - |
+          cd {{.TMP_INSTALL_DIR}}
+          rm -f {{.TMP_INSTALL_DIR}}/nonpriv_users.txt
+          rm -f {{.TMP_INSTALL_DIR}}/processes_to_restart.txt
+
+          #
+          # For guided installs, we currently support `php-fpm`, `apache`, and `nginx`.
+          # There is a reason for the order.  According our install script:
```

Comment:
If "install script" is a reference to the deploy config, the decision made there was merely a convenience since we have control of that environment. It isn't something we should rely on or mention here.

Reply:
TIL. I didn't know we had any such detection logic in `newrelic-install.sh`. I don't think the reference needs to be removed.

Reply:
I also put `nginx` because I thought that was the focus of the current [MMF](https://github.com/newrelic/newrelic-php-agent/issues/93).  Is that incorrect?

```
For users of Debian or RHEL derived systems, running PHP under apache or fpm/nginx, the New Relic Open Installer library can be used to easily install and configure the PHP Agent on their systems.

This epic targets the initial buildout of supporting this, with a focus on FPM+nginx under Debian. Additional target environments will be supported in the subsequent epic.
```